test/stdlib/Timeout_test.rb

```diff
@@ -0,0 +1,41 @@
+require_relative "test_helper"
+require "timeout"
+require "bigdecimal"
+
+class TimeoutSingletonTest < Test::Unit::TestCase
+  include TypeAssertions
+
+  library "timeout"
+  testing "singleton(::Timeout)"
+
+  class TimeoutTestException < Exception; end # exception class for test
+
+  def test_timeout
+    proc = Proc.new { |sec| sec * sec }
+    assert_send_type  "(::Integer sec) { (::Integer sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 5, &proc
+    assert_send_type  "(::Float sec) { (::Float sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 1.2, &proc
+    assert_send_type  "(::Rational sec) { (::Rational sec) -> untyped } -> untyped",
+                      Timeout, :timeout, Rational(5, 3), &proc
+    assert_send_type  "(::BigDecimal sec) { (::BigDecimal sec) -> untyped } -> untyped",
+                      Timeout, :timeout, BigDecimal("1.123456789123456789"), &proc
+
+    hard_process = Proc.new { _calc_pi }
+    refute_send_type  "(::Numeric sec) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, &hard_process
+    refute_send_type  "(::Numeric sec, singleton(Exception) klass) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, TimeoutTestException, &hard_process
+    refute_send_type  "(::Numeric sec, singleton(Exception) klass, String message) { (::Numeric sec) -> untyped } -> untyped",
+                      Timeout, :timeout, 0.001, TimeoutTestException, "timeout test error", &hard_process
```

Comment:
I'm not sure that this `refute_send_type` usage is appropriate. 

I guess `refute_send_type` is designed to check an unacceptable type. For example, `Timeout.timeout('1')` is invalid with the type, so checking `Timeout.timeout('1')` call with `refute_send_type` is the designed usage.

But these tests, such as `Timeout.timeout(0.001){...}`, is acceptable with the type. It just checks `Timeout.timeout` raises an error, but the raised error is an "expected" error. So the `refute_send_type` calls just suppress the error. I think `refute_send_type` is not appropriate in this case.


related: #588

Reply:
Thank you for your advice! I can understand.
I fixed it in 083beb9 . Would you please review again.